template-parts/post-list-filters.php

```diff
@@ -0,0 +1,153 @@
+							'shiro'
+						)
+					),
+					esc_html( $first_result ),
+					esc_html( $last_result ),
+					esc_html( $total_results )
+				);
+			}
+			?>
+
+		<button class="action-button post-list-filter__toggle">
+			<span class="post-list-filter__toggle__message--hide"><?php echo esc_html__( 'Hide filters', 'shiro' ); ?></span>
+			<span class="post-list-filter__toggle__message--show">
+			<?php
+
+			printf( esc_html__( 'Show filters', 'shiro' ) );
+
+			if ( isset( $_GET['post_list_filters_nonce'] ) && wp_verify_nonce( sanitize_text_field( $_GET['post_list_filters_nonce'] ), 'post_list_filters' ) ) {
+				$filter_count = 0;
+
+				// Search term.
+				if ( ! empty( $query_var_search_term ) ) {
+					$filter_count++;
+				}
+
+				// Date interval.
+				if ( ! empty( $query_var_date_from ) || ! empty( $query_var_date_to ) ) {
+					$filter_count++;
+				}
+
+				// Categories.
+				$filter_count += count( $query_var_categories );
+
+				if ( $filter_count > 0 ) {
+					echo ' <em>';
+					/* translators: 1. how many filters were applied */
+					printf( esc_html__( '(%s applied)', 'shiro' ), esc_html( $filter_count ) );
+					echo '</em>';
+				}
+			}
+			?>
+			</span>
+		</button>
+
+	</div>
+
+	<form method="get" class="post-list-filter__form">
+
+		<?php wp_nonce_field( 'post_list_filters', 'post_list_filters_nonce' ); ?>
+
+		<div class="post-list-filter__container mw-980">
+
+			<div class="filter-by-text">
+				<h5>
+					<?php printf( esc_html__( 'Filter by text', 'shiro' ) ); ?>
+					<?php if ( ! empty( $query_var_search_term ) ) : ?>
+						&nbsp;<em>(<?php printf( esc_html__( 'applied', 'shiro' ) ); ?>)</em>
+					<?php endif; ?>
+				</h5>
+				<div class="search-text-input-button">
+					<input type="text" name="s" value="<?php echo esc_attr( $query_var_search_term ); ?>">
+				</div>
+			</div>
+
+			<div class="filter-by-date">
+				<h5>
+					<?php printf( esc_html__( 'Filter by date', 'shiro' ) ); ?>
+					<?php if ( ! empty( $query_var_date_from ) || ! empty( $query_var_date_to ) ) : ?>
+						&nbsp;<em>(<?php printf( esc_html__( 'applied', 'shiro' ) ); ?>)</em>
+					<?php endif; ?>
+				</h5>
+				<div class="filter-date-inputs-container">
+					<input type="date" name="date_from" placeholder="From" value="<?php echo esc_attr( $query_var_date_from ); ?>">
+					<input type="date" name="date_to" placeholder="To" value="<?php echo esc_attr( $query_var_date_to ); ?>">
+					<button type="button" class="action-button action-button--clear" id="button-reset-date-filters"><?php printf( esc_html__( 'Reset', 'shiro' ) ); ?></button>
+				</div>
+			</div>
+
+			<div class="filter-by-category">
+				<?php
+				$categories = get_categories();
+				foreach ( $categories as $category ) {
+					$category_display = ( $category->parent == 0 )
+						? $category->name
+						: get_category_parents( $category->parent, false, ' > ', false ) . $category->name;
+
+					$categories_array[ $category->slug ] = $category_display;
```

Comment:
So this is interesting -- I didn't know that you could implicitly create an array like this in PHP! What I mean is, that if you do this,
```php
<?php

$arr['something'] = 'foo';
print_r( $arr );
```
you get
```
Array
(
    [something] => foo
)
```
even if `$arr` wasn't previously defined. So, this line is creating `$categories_array` as an array implicitly.

I'd recommend making this implicit behavior more explicit, by declaring `$categories_array = [];` before the `foreach` loop.

I'd also suggest moving this array computation up to the initial PHP block at the top of the template partial, along with the other template variables.